src/test/java/com/connexta/search/SearchITest.java

```diff
@@ -97,309 +90,485 @@ public URL solrUrl() throws MalformedURLException {
-  @ParameterizedTest
-  @ValueSource(strings = {"", "{}", "{ \"\": \"text\"}"})
-  public void testStoringInvalidCst(String contents) throws IOException {
-    // given
-    final String indexEndpointUrl = INDEX_ENDPOINT_BASE_URL + "00067360b70e4acfab561fe593ad3f7a";
-    // when indexing IRM
-    ResponseEntity<String> response =
-        restTemplate.exchange(
-            indexEndpointUrl, HttpMethod.PUT, createIndexRequest(contents), String.class);
-
-    // then
-    assertThat(response.getStatusCode(), is(HttpStatus.BAD_REQUEST));
-  }
+  @Test
+  void testContextLoads() {}
 
   @Test
-  public void testStoreMetadataCstWhenSolrIsEmpty() throws Exception {
-    // given
-    final String queryKeyword = "Winterfell";
-    final String indexEndpointUrl = INDEX_ENDPOINT_BASE_URL + "00067360b70e4acfab561fe593ad3f7a";
-    final String irmLocation =
-        String.format(
-            "%s/%s/%s",
-            datasetRetrieveEndpoint, "00067360b70e4acfab561fe593ad3f7a", IRM_PATH_SEGMENT);
-
-    // when indexing IRM
-    restTemplate.put(
-        indexEndpointUrl,
-        createIndexRequest(
-            "{ \"ext.extracted.text\" : \""
-                + ("All the color had been leached from "
-                    + queryKeyword
-                    + " until only grey and white remained")
-                + " \" }"));
-
-    // then
-    final URIBuilder queryUriBuilder = new URIBuilder();
-    queryUriBuilder.setPath(QueryController.URL_TEMPLATE);
-    queryUriBuilder.setParameter(
-        "q", String.format("%s LIKE '%s'", CONTENTS_ATTRIBUTE, queryKeyword));
-    assertThat(
-        (List<String>) restTemplate.getForObject(queryUriBuilder.build(), List.class),
-        hasItem(irmLocation));
+  void testIndexWhenSolrIsEmpty() throws Exception {
+    // given stub store server
+    final String keyword = "Winterfell";
+    storeMockWebServer.enqueue(
+        new MockResponse()
+            .setBody(
+                String.format(
+                    "{ \"%s\" : \"All the color had been leached from %s until only grey and white remained.\" }",
+                    EXT_EXTRACTED_TEXT, keyword))
+            .setResponseCode(200));
+    final String datasetId = "00067360b70e4acfab561fe593ad3f7a";
+    final URI irmUri = storeMockWebServer.url(String.format("/dataset/%s/irm", datasetId)).uri();
+
+    // when index
+    webTestClient
+        .put()
+        .uri(IndexController.URL_TEMPLATE, datasetId)
+        .header(IndexController.ACCEPT_VERSION_HEADER_NAME, indexApiVersion)
+        .bodyValue((new IndexRequest().irmLocation(irmUri)))
+        .exchange()
+        .expectStatus()
+        .isOk();
+
+    // then verify GET request to irmUri
+    final RecordedRequest getIrmRequest = storeMockWebServer.takeRequest();
+    assertThat(getIrmRequest.getMethod(), is(HttpMethod.GET.name()));
+    assertThat(getIrmRequest.getPath(), is(String.format("/dataset/%s/irm", datasetId)));
+
+    // and verify query returns irmUri
+    webTestClient
```

Comment:
There is a guarantee that indexRepository.save has been called at this point in the test. I think we can wait to address any timing issues until we encounter them. I think we would have seen timing issues already because we save and immediately query in a few existing tests.